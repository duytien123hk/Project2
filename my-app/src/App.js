import { Route, Routes } from "react-router-dom";
import "./App.css";
import DashBoard from "./components/DashBoard";
import PollDetail from "./components/PollDetail";
import Login from "./components/Login";
import CreatePoll from "./components/CreatePoll";
import LeaderBoard from "./components/LeaderBoard";
import { useEffect } from "react";
import { handleInitialData } from "./actions/shared";
import { connect } from "react-redux";
import Error from "./components/Error";
import HandleRoute from "./components/HandleRoute";

function App({ dispatch }) {
  useEffect(() => {
    dispatch(handleInitialData());
  }, []);

  return (
    <div>
      <Routes>
        <Route path="login" exac element={<Login />}></Route>
        <Route
          path="/"
          element={
            <HandleRoute>
              <DashBoard />
            </HandleRoute>
          }
        ></Route>
        <Route
          path="/questions/:question_id"
          element={
            <HandleRoute>
              <PollDetail />
            </HandleRoute>
          }
        ></Route>
        <Route
          path="/leaderboard"
          element={
            <HandleRoute>
              <LeaderBoard />
            </HandleRoute>
          }
        ></Route>
        <Route
          path="/add"
          element={
            <HandleRoute>
              <CreatePoll />
            </HandleRoute>
          }
        ></Route>
        <Route path="/error" element={<Error />}></Route>
      </Routes>
    </div>
  );
}

export default connect()(App);
