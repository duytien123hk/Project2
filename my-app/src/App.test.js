import { render } from "@testing-library/react";
import * as React from "react";
import App from "./App";
import { store } from "./store";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

describe("App", () => {
  it("render app", async () => {
    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>
    );

    expect(component).toMatchSnapshot();
  });
});
