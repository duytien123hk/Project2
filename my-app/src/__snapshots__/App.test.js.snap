// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render app 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="container-login"
        >
          <form
            class="login-box"
          >
            <h3
              class="login-header"
            >
              Login
            </h3>
            <div
              class="input-box"
            >
              <input
                autocomplete="username"
                class="input-field"
                data-testid="username"
                placeholder="Username"
                type="text"
                value="sarahedo"
              />
            </div>
            <div
              class="input-box"
            >
              <input
                autocomplete="current-password"
                class="input-field"
                data-testid="password"
                placeholder="Password"
                type="password"
                value="123"
              />
            </div>
            <button
              class="btn"
              data-testid="submit"
            >
              Submit
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="container-login"
      >
        <form
          class="login-box"
        >
          <h3
            class="login-header"
          >
            Login
          </h3>
          <div
            class="input-box"
          >
            <input
              autocomplete="username"
              class="input-field"
              data-testid="username"
              placeholder="Username"
              type="text"
              value="sarahedo"
            />
          </div>
          <div
            class="input-box"
          >
            <input
              autocomplete="current-password"
              class="input-field"
              data-testid="password"
              placeholder="Password"
              type="password"
              value="123"
            />
          </div>
          <button
            class="btn"
            data-testid="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
